schemaregistry/config.go

```diff
@@ -0,0 +1,66 @@
+package schemaregistry
+
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Config is used to pass multiple configuration options to the Schema Registry client.
+type Config struct {
+	// SchemaRegistryURL determines the URL of Schema Registry
+	SchemaRegistryURL string
+
+	// BasicAuth keys
+	BasicAuthUserInfo          string
+	BasicAuthCredentialsSource string
+
+	// Sasl keys
+	SaslMechanism string
+	SaslUsername  string
+	SaslPassword  string
+
+	// Ssl Keys
+	SslCertificationLocation       string
+	SslKeyLocation                 string
+	SslCaLocation                  string
+	SslDisableEndpointVerification bool
+
+	// Timeouts
+	ConnectionTimeoutMs int
+	RequestTimeoutMs    int
+}
+
+// NewConfig returns a new configuration instance with sane defaults.
+func NewConfig(url string) *Config {
```

Comment:
I expect you considered having a parameterless constructor as well, and went with having url (the only thing required) as a parameter to minimize verbosity.

```
client, err := schemaregistry.NewClient(url)
```

vs

```
srConf := schemaregistry.NewConfig()
srConf.SchemaRegistryURL = url
client, err := schemaregistry.NewClient(srConf)
```

the argument against this is extensibility. right now there is only one required parameter, but what if in the future there is more? i realize it's still extensible in that you just set the other required parameter after you construct it, but then it's inconsistent. i also realize it's unlikely we'll ever have another required parameter. my argument is more theoretical / philosophical. i'm that kind of person. which is why i prefer going with the parameterless version. but i understand that other people see the world differently.

if go had method overloads or optional parameters, my sense here on what is best would probably be different.